tests/Properties.hs

```diff
@@ -76,20 +77,18 @@ prop_unsafeIndexBB xs =
 prop_bijectionBB  (Char8 c) = (P.w2c . P.c2w) c == id c
 prop_bijectionBB'        w  = (P.c2w . P.w2c) w == id w
 
-prop_head2BB xs    = (not (null xs)) ==> head xs   == (P.unsafeHead . P.pack) xs
+prop_unsafeHead x = maybe True ((== P.unsafeHead x) . fst) (List.uncons (P.unpack x))
 
-prop_tail1BB xs    = (not (null xs)) ==> tail xs    == (P.unpack . P.unsafeTail. P.pack) xs
+prop_unsafeTail x = maybe True ((== P.unpack (P.unsafeTail x)) . snd) (List.uncons (P.unpack x))
```

Comment:
I think these changes have significantly reduced readability of the tests. I also think that using `head` and `tail` in tests is an OK practice.